### DISCUSSION TITLE
Use onNotificationClick / onClick Event in embedded Package

### DISCUSSION THREAD

**Discussion by MarcKoester** [magicbell/community#134] [Questions] [answered]
We want to use magicbell inside an angular application. We want to customize the onNotificationClick behaviour. (Using angular router for internal links).
It isn't documented but you can give the `renderWidget` function a `onNotificationClick` function which gets called.
But we can't stop magicbell to call [window.open](https://github.com/magicbell-io/magicbell-js/blob/81c52209ed1eec0b84e6a8fbcd187dd0a5f897ea/packages/react/src/components/ClickableNotification/eventHandlers.ts#L9C2-L9C2)
Is there a way to stop magicbell to call window.open?

It could be fixed by adding a return in [this line](https://github.com/magicbell-io/magicbell-js/blob/81c52209ed1eec0b84e6a8fbcd187dd0a5f897ea/packages/embeddable/src/components/FloatingFrame/FloatingFrame.tsx#L40). Then our `onNotificationClick` could return false and `window.open` wouldn't be called because of [this early return]( https://github.com/magicbell-io/magicbell-js/blob/81c52209ed1eec0b84e6a8fbcd187dd0a5f897ea/packages/react/src/components/ClickableNotification/ClickableNotification.tsx#L44).

**Comment by unamashana** [comment]
Hey @MarcKoester sorry about that. Can you please confirm the versions of the MagicBell SDK/packages you are using? Thanks!

**Reply by MarcKoester** [reply]
Sure we are using @magicbell/embeddable v3.2.8

**Reply by smeijer** [reply, marked as answer]
Can you please upgrade to 3.3.1, and let us know if the issue persists? Returning `false` should prevent opening the actionUrl.
